lib/Driver/ToolChains.cpp

```diff
@@ -1044,6 +1044,24 @@ addLinkSanitizerLibArgsForDarwin(const ArgList &Args,
                      /*AddRPath*/ true, TC);
 }
 
+static void
+addLinkFuzzerLibArgsForDarwin(const ArgList &Args,
+                       ArgStringList &Arguments,
+                       const ToolChain &TC) {
+
+  // libFuzzer requires C++.
+  Arguments.push_back("-lc++");
+
+  // Link libfuzzer.
+  SmallString<128> Dir;
+  getRuntimeLibraryPath(Dir, Args, TC);
+  llvm::sys::path::remove_filename(Dir);
+  llvm::sys::path::append(Dir, "llvm", "libLLVMFuzzer.a");
```

Comment:
> There's no such thing as an "LLVM resource dir" in a normal installation of Clang

I am not sure what do you mean. There's a `lib` dir, which includes all kind of libraries shipped with LLVM, and this pull request also adds a symlink to it.

> (Also, http://llvm.org/docs/LibFuzzer.html talks about "libFuzzer.a" rather than "libLLVMFuzzer.a". Are they equivalent?)

They refer to the compilation unit produced manually via a shell script.
This pull request grabs an archive file produced by CMake.